src/main/java/bio/terra/service/duos/DuosService.java

```diff
@@ -148,13 +152,97 @@ private void updateLastSynced(DuosFirecloudGroupModel firecloudGroup, Instant la
     try {
       duosDao.updateFirecloudGroupLastSyncedDate(id, lastSyncedDate);
     } catch (CannotSerializeTransactionException ex) {
-      String msg =
-          "Members of Firecloud group %s were updated, "
-              + "but could not update DUOS Firecloud Group record %s with last_synced_date %s"
-                  .formatted(firecloudGroup.getFirecloudGroupName(), id, lastSyncedDate);
+      String message =
+          firecloudGroup.getFirecloudGroupEmail()
+              + " members were updated, "
+              + "but an error occurred when updating its database record with last_synced_date "
+              + lastSyncedDate;
       String errorDetail =
           "This indicates that another process was updating the database row at the same time";
-      throw new DuosFirecloudGroupUpdateConflictException(msg, ex, List.of(errorDetail));
+      throw new DuosFirecloudGroupUpdateConflictException(message, ex, List.of(errorDetail));
+    }
+  }
+
+  /**
+   * For all DUOS dataset IDs registered in TDR, force a sync of their Firecloud group members. A
+   * failure to sync any Firecloud group(s) should not block attempts to sync the rest.
+   *
+   * @return DUOS Firecloud groups whose members were successfully synced, and any errors occurred
+   *     which may have interfered with syncing
+   */
+  public DuosFirecloudGroupsSyncResponse syncDuosDatasetsAuthorizedUsers() {
+    List<ErrorModel> errors = new ArrayList<>();
```

Comment:
This list is updated on different threads but `ArrayList` isn't thread safe. I think you want
```suggestion
    List<ErrorModel> errors = Collections.synchronizedList(new ArrayList<>());
```

Reply:
Thanks, I do!